### PR TITLE
Feature/multiple deliveries in a day

### PR DESCRIPTION
**Pull request by martijnlibaro**

### CODE REVIEW

File: app/Actions/ChooseRunner.php [hunk 1]
```diff
@@ -9,22 +9,31 @@
...
 use Illuminate\Support\Facades\Mail;
 
 final class ChooseRunner
 {
     private Company $company;
+    private $user;
+    private $addTomorrow;
 
-    public function __construct(Company $company)
+    public function __construct(Company $company, $user = null, $addTomorrow = false)
```
Comment (JennisDev) [inline app/Actions/ChooseRunner.php:22]: Typecasting voor user en boolean

File: app/Actions/ChooseRunner.php [hunk 2]
```diff
@@ -71,6 +80,17 @@ private function setCompany($company)
         return $this;
     }
 
+    /**
+     * @param $user
+     * @return $this
+     */
+    private function setUser($user)
```
Comment (JennisDev) [inline app/Actions/ChooseRunner.php:87]: Typecasting voor user

File: app/Actions/ChooseRunner.php [hunk 3]
```diff
@@ -9,22 +9,31 @@
...
 use App\Models\Company;
 use App\Models\Order;
+use App\Models\User;
 use Carbon\Carbon;
 use Illuminate\Support\Facades\Mail;
 
 final class ChooseRunner
 {
     private Company $company;
+    private $user;
```
Comment (RosiersRobin) [inline app/Actions/ChooseRunner.php:19]: Strict model binding for variable; `private User $user;`

File: app/Actions/ChooseRunner.php [hunk 4]
```diff
@@ -9,22 +9,31 @@
...
 
-    public function __construct(Company $company)
+    public function __construct(Company $company, $user = null, $addTomorrow = false)
     {
         $this->setCompany($company);
+        $this->setUser($user);
+        $this->addTomorrow = $addTomorrow;
     }
 
     public function execute()
```
Comment (RosiersRobin) [inline app/Actions/ChooseRunner.php:29]: return type `execute(): this`

File: app/Actions/ChooseRunner.php (hunk 2 above)
Comment (RosiersRobin) [inline app/Actions/ChooseRunner.php:87]: Returntypes

File: app/Actions/ChooseRunner.php [hunk 5]
```diff
@@ -71,6 +80,17 @@ private function setCompany($company)
...
+    private function setUser($user)
+    {
+        $this->user = $user;
+
+        return $this;
+    }
+
     private function sendMission($victim, $orders)
     {
         Mail::to($victim->email)->bcc('jennis@libaro.be')->send(new InformVictim($orders, $victim));
```
Comment (RosiersRobin) [inline app/Actions/ChooseRunner.php:96]: You should remove a static bcc...

File: app/Http/Controllers/OrderController.php [hunk 6]
```diff
@@ -152,6 +152,17 @@ public function removeProduct(RemoveRequest $request): JsonResponse
         ]);
     }
 
+    public function assignToMe(){
```
Comment (RosiersRobin) [inline app/Http/Controllers/OrderController.php:155]: returntypes

File: app/Models/Order.php [hunk 7]
```diff
@@ -43,11 +43,19 @@ public function product(): BelongsTo
         return $this->belongsTo(Product::class);
     }
 
-    public static function getOrders(Company $company, $date)
+    public static function getOrders(Company $company, $date, $addTomorrow = false)
```
Comment (RosiersRobin) [inline app/Models/Order.php:46]: typehints

File: app/Models/Order.php [hunk 8]
```diff
@@ -43,11 +43,19 @@ public function product(): BelongsTo
...
-            ->where('date', '<=', (clone $date)->endOfDay())
+        $from = (clone $date)->startOfDay();
+        $to = (clone $date)->endOfDay();
+        if($addTomorrow){
+            $to->addDay();
+        }
+
+        return self::query()
+            ->where('company_id', $company->id)
+            ->where('date', '>=', $from)
```
Comment (RosiersRobin) [inline app/Models/Order.php:56]: 2 wheres can be in one wherebetween. Is faster than 2 wheres too.

File: app/Actions/ChooseRunner.php [hunk 9]
```diff
@@ -9,22 +9,36 @@
...
 use App\Models\Company;
 use App\Models\Order;
+use App\Models\User;
 use Carbon\Carbon;
 use Illuminate\Support\Facades\Mail;
 
 final class ChooseRunner
 {
     private Company $company;
+    private ?User $user;
```
Comment (JennisDev) [inline app/Actions/ChooseRunner.php:19]: questionmark intended?

**Pull request merged**